Implement delete, update, insert, truncate

mebelousov opened the pull request:
`updater_clause` is not perfect, others are good.
Docs:
https://www.postgresql.org/docs/current/sql-delete.html
https://www.postgresql.org/docs/current/sql-insert.html
https://www.postgresql.org/docs/current/sql-update.html
https://www.postgresql.org/docs/current/sql-truncate.html

tshprecher commented:
This is great! I'll take a look at it tonight. I'll have to check against the documentation, but I trust it's mostly right. 

This fills an important gap in the grammar. Thank you!

tshprecher commented on antlr4/PostgreSQLParser.g4:
```diff
@@ -1316,6 +1333,17 @@ for_clause
+    ;
+
+returning_clause
+    : RETURNING ((column_name_=expr (AS? name_)?) | STAR)
```
nit: assign a label to `name_` -> `output_name=name_`. We may not do a great job at this across the grammar, but I was planning on doing some normalization once I'm happy with the correctness of the grammar as defined by the success rates of the tests.

It's generally hard to get things consistent at this stage of the development because there are so many rules.

mebelousov replied:
Good idea. 
Done!

tshprecher commented on antlr4/PostgreSQLParser.g4:
```diff
@@ -1316,6 +1333,17 @@ for_clause
     ;
 
+updater_clause
+    : (expr EQUAL expr (COMMA expr EQUAL expr)*)
```
Based on my reading of the grammar definition, I think we want the each of these three cases to be list-able. In other words, I think it should be possible to have `SET blah=1, (a, b, c) = (1, 2, 3), foo=2`. 

I think that's what you're alluding to in your comment to updater not being perfect. That's fine. We can iterate from here.

tshprecher commented on antlr4/PostgreSQLParser.g4:
```diff
@@ -1081,7 +1084,16 @@ import_foreign_schema_stmt
+        (OPEN_PAREN column_name CLOSE_PAREN)?
+        (ON CONSTRAINT column_name)?
+        where_clause?
+        ((DO NOTHING)|(DO UPDATE SET updater_clause where_clause?))
```
Nice dual use of `updater_clause` here alongside update_stmt! 👍

tshprecher commented on antlr4/PostgreSQLParser.g4:
```diff
@@ -1198,15 +1210,20 @@ show_stmt
 
 truncate_stmt
-    : todo_implement
+    : TRUNCATE TABLE? ONLY? table_name_ STAR? (COMMA ONLY? identifier_list)?
```
I think the second `ONLY?` can be removed here.

mebelousov replied:
See [src/test/resources/sql/truncate/a31f29bb.sql](https://github.com/tshprecher/antlr_psql/blob/master/src/test/resources/sql/truncate/a31f29bb.sql)
```TRUNCATE ONLY trunc_fb, ONLY trunc_fa```

mebelousov commented:
I added `updater_expr`. Also I removed from it `OPEN_PAREN name_list CLOSE_PAREN EQUAL OPEN_PAREN select_stmt CLOSE_PAREN` because of `expr` contains ` OPEN_PAREN select_stmt CLOSE_PAREN`
